src/codegate/types/openai/_request_models.py

```diff
@@ -339,9 +339,9 @@ def last_user_message(self) -> tuple[Message, int] | None:
 
     def last_user_block(self) -> Iterable[tuple[Message, int]]:
         for idx, msg in enumerate(reversed(self.messages)):
-            if isinstance(msg, UserMessage):
+            if isinstance(msg, (UserMessage, ToolMessage)):
```

Comment:
This is to be able to extract bad packages from files read by the tools, see the unit test for an example I grabbed from OpenInterpreter session. I totally forgot that tool call results can contain bad packages...